dbt/adapters/athena/impl.py

```diff
@@ -583,7 +582,7 @@ def _get_one_table_for_non_glue_catalog(
                     "column_comment": col.get("Comment", ""),
                 },
             }
-            for idx, col in enumerate(table["Columns"] + table.get("PartitionKeys", []))
+            for idx, col in enumerate(table.get("Columns", []) + table.get("PartitionKeys", []))
```

Comment:
for the sake of knowledge, what are the scenarios where Columns is not there?

Reply:
TBH "Columns" key is almost absent for TableTypeDef class. I don't use external catalogs, so I'm not sure if it works at all )

Reply:
just that fact that there is no Columns key in mentioned class:
```
TableTypeDef = TypedDict(
    "TableTypeDef",
    {
        "Name": str,
        "DatabaseName": NotRequired[str],
        "Description": NotRequired[str],
        "Owner": NotRequired[str],
        "CreateTime": NotRequired[datetime],
        "UpdateTime": NotRequired[datetime],
        "LastAccessTime": NotRequired[datetime],
        "LastAnalyzedTime": NotRequired[datetime],
        "Retention": NotRequired[int],
        "StorageDescriptor": NotRequired[StorageDescriptorOutputTypeDef],
        "PartitionKeys": NotRequired[List[ColumnOutputTypeDef]],
        "ViewOriginalText": NotRequired[str],
        "ViewExpandedText": NotRequired[str],
        "TableType": NotRequired[str],
        "Parameters": NotRequired[Dict[str, str]],
        "CreatedBy": NotRequired[str],
        "IsRegisteredWithLakeFormation": NotRequired[bool],
        "TargetTable": NotRequired[TableIdentifierTypeDef],
        "CatalogId": NotRequired[str],
        "VersionId": NotRequired[str],
        "FederatedTable": NotRequired[FederatedTableTypeDef],
        "ViewDefinition": NotRequired[ViewDefinitionTypeDef],
        "IsMultiDialectView": NotRequired[bool],
    },
)
```

Reply:
check [this](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/glue/client/get_table.html), **Columns** is in `StorageDescriptor `, as we do in [989b6b80bb7a2ce0abb5ace7545a215a2f79/dbt/adapters/athena/impl.py#L563](https://github.com/dbt-athena/dbt-athena/blob/3c66989b6b80bb7a2ce0abb5ace7545a215a2f79/dbt/adapters/athena/impl.py#L563), I believe that we should use `table.get("StorageDescriptor", {}].get("Columns", [])`

Reply:
so I just don't want to get an error if I start using such external catalogs. And the error is possible in this case (just based on python code)

Reply:
> check [this](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/glue/client/get_table.html), **Columns** is in `StorageDescriptor `, as we do in [989b6b80bb7a2ce0abb5ace7545a215a2f79/dbt/adapters/athena/impl.py#L563](https://github.com/dbt-athena/dbt-athena/blob/3c66989b6b80bb7a2ce0abb5ace7545a215a2f79/dbt/adapters/athena/impl.py#L563), I believe that we should use `table.get("StorageDescriptor", {}].get("Columns", [])`

I thought about that as well, but it worked somehow before...I just didn't want to break it now